### DISCUSSION TITLE
Race conditions between space initialization and space writes make WAL corrupted. Need tarantool instance lifecycle states improved.

### DISCUSSION THREAD

**Discussion by extde** [tarantool/tarantool#9325] [Ideas]
Got an issue with corrupted WAL log due to race conditions between space initialization and space writes during tarantool instance startup. 
Looks like a bug because WAL is corrupted, restart is not possible.
See details here - https://stackoverflow.com/a/77408135/2412036

[tarantool race conditions example.zip](https://github.com/tarantool/tarantool/files/13237875/tarantool.race.conditions.example.zip)

It would be nice to have clear set of tarantool instance lifecycle states, including ones for spaces initialization, business logic initialization, instance termination, intermediate states (like service overloaded), etc.
It would be nice if this set would be customizable.

**Comment by sergepetrenko** [comment]
Judging by your description on stackoverflow this looks like a bug and needs investigation. https://github.com/tarantool/tarantool/issues/9330
